kotlin/workflow-ui-android/src/test/java/com/squareup/workflow/ui/WorkflowRunnerViewModelTest.kt

```diff
@@ -39,18 +52,83 @@ class WorkflowRunnerViewModelTest {
     assertThat(runner.getLastSnapshotForTest()).isEqualTo(snapshot2)
   }
 
-  @Test fun hostCancelledOnCleared() {
+  @Test fun hostCancelledOnResultAndNoSooner() {
     var cancelled = false
     scope.coroutineContext[Job]!!.invokeOnCompletion { e ->
-      if (e is CancellationException) cancelled = true
+      if (e is CancellationException) {
+        cancelled = true
+      } else throw AssertionError(
+          "Expected ${CancellationException::class.simpleName}", e
+      )
     }
-    val runner = WorkflowRunnerViewModel(viewRegistry, emptyFlow(), emptyFlow(), scope)
+    val runner = WorkflowRunnerViewModel(scope, emptyFlow(), flowOf("fnord"), viewRegistry)
 
     assertThat(cancelled).isFalse()
-    runner.output.test()
-    assertThat(cancelled).isFalse()
+    val tester = runner.result.test()
+    assertThat(cancelled).isTrue()
+    tester.assertComplete()
+    assertThat(tester.values()).isEqualTo(listOf("fnord"))
+  }
 
+  @Test fun hostCancelledOnCleared() {
+    var cancelled = false
+    scope.coroutineContext[Job]!!.invokeOnCompletion { e ->
+      if (e is CancellationException) {
+        cancelled = true
+      } else throw AssertionError(
+          "Expected ${CancellationException::class.simpleName}", e
+      )
+    }
+    val runner = WorkflowRunnerViewModel(scope, emptyFlow(), emptyFlow(), viewRegistry)
+
+    assertThat(cancelled).isFalse()
+    val tester = runner.result.test()
     runner.clearForTest()
     assertThat(cancelled).isTrue()
+    tester.assertComplete()
+    tester.assertNoValues()
+  }
+
+  @Test fun resultDelivered() {
+    val outputs = BroadcastChannel<String>(1)
+    val runner = Workflow
+        .stateless<Unit, String, Unit> {
+          onWorkerOutput(outputs.asWorker()) { emitOutput(it) }
+          Unit
+        }
+        .run()
+
+    val tester = runner.result.test()
+
+    tester.assertNotComplete()
+    runBlocking { outputs.send("fnord") }
+    tester.assertComplete()
```

Comment:
@zach-klippenstein Halp, this assertion fails and I can't figure out why. Especially confusing b/c the less-integrationy `hostCancelledOnResultAndNoSooner` test above passes.